pkg/dogstatsd/intern.go

```diff
@@ -0,0 +1,25 @@
+package dogstatsd
+
+type stringInterner struct {
+	strings map[string]string
+	maxSize int
+}
+
+func newStringInterner(maxSize int) *stringInterner {
+	return &stringInterner{
+		strings: make(map[string]string),
+		maxSize: maxSize,
+	}
+}
+
+func (i *stringInterner) LoadOrStore(key []byte) string {
+	if len(i.strings) >= i.maxSize {
+		i.strings = make(map[string]string)
```

Comment:
As we are completely reseting the cache here (right?), I'm wondering if we should throw a DEBUG or something here, or even a telemetry metric, to observe if it's happening frequently in order to tweak the `maxSize` or to know that we have to think about a more complex solution. WDYT? Feel free to discard.
Had you the chance to bench it?